File: tools/improve/tests/devstral-small-2/maudec/py/nat-funs.py
#
#	<comment from="devstral-small-2">
#	I've simplified and improved the efficiency of the code while preserving the semantics and original function signatures. Here's the optimized version:
#	</comment>
#

def fib(a1: int) -> int:
    if a1 == 0:
        return 0
    elif a1 == 1:
        return 1
    else:
        return fib(a1 - 2) + fib(a1 - 1)


def _u0021(a1: int) -> int:
    if a1 == 0:
        return 1
    else:
        return a1 * _u0021(a1 - 1)


def u0024gcd(M: int, N: int) -> int:
    if N == 0:
        return M
    else:
        return u0024gcd(N, M % N)

File: tools/improve/tests/devstral-small-2/maudec/rust/nat-funs.rs
//
//	<comment from="devstral-small-2">
//	I'll simplify and improve the efficiency of the Rust code while preserving the semantics and original function signatures.
//	</comment>
//

fn fib(a1: i32) -> i32 {
    match a1 {
        0 => 0,
        1 => 1,
        n if n >= 2 => fib(n - 2) + fib(n - 1),
        _ => panic!("case not covered"),
    }
}

fn _u0021(a1: i32) -> i32 {
    match a1 {
        0 => 1,
        n if n >= 1 => n * _u0021(n - 1),
        _ => panic!("case not covered"),
    }
}

fn u0024gcd(m: i32, n: i32) -> i32 {
    match n {
        0 => m,
        n => u0024gcd(n, m % n),
    }
}